### PR TITLE
implement aks gpu instance profile

### PR DESCRIPTION
**Pull request by alexeldeib**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
   - this fails but not on anything related to my diff.
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
aks

**Comment by FumingZhang** [comment]
It seems that the live test pipeline failed on a removed test case `test_aks_nodepool_add_with_workload_runtime`, but `test_aks_nodepool_add_with_gpu_instance_profile` is not tested? Could you please queue another test? If the newly added option needs feature registration,  you may exclude this test case in [ext_matrix_default.json](https://github.com/alexeldeib/azure-cli-extensions/blob/ace/mig/src/aks-preview/azcli_aks_live_test/configs/ext_matrix_default.json).

**Comment by alexeldeib** [comment]
ah sorry, I mixed up the tests with this and #3896

**Comment by alexeldeib** [comment]
I'll exclude the live test due to feature registration, unfortunately same issue applies to #3896

File: src/aks-preview/azext_aks_preview/_params.py [hunk 1]
```diff
@@ -140,6 +142,7 @@ def load_arguments(self, _):
         c.argument('enable_secret_rotation', action='store_true')
         c.argument('assign_kubelet_identity', type=str, validator=validate_assign_kubelet_identity)
         c.argument('disable_local_accounts', action='store_true')
+        c.argument('gpu_instance_profile', arg_type=get_enum_type([CONST_GPU_INSTANCE_PROFILE_MIG1_G, CONST_GPU_INSTANCE_PROFILE_MIG2_G, CONST_GPU_INSTANCE_PROFILE_MIG3_G, CONST_GPU_INSTANCE_PROFILE_MIG4_G, CONST_GPU_INSTANCE_PROFILE_MIG7_G]))
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:145]: May consider using a variable to wrap these options?

**Review by FumingZhang — APPROVED** [review]

**Comment by alexeldeib** [comment]
@FumingZhang anything we're waiting on here for merge?

**Comment by zhoxing-ms** [comment]
@alexeldeib Please write this change in to the `HISTORY.rst`

By the way, do you need to release a new version for these changes? If so, please also update the `setup.py`

**Comment by zhoxing-ms** [comment]
@alexeldeib Could you please address those conflicts?

**Pull request merged**